yamllint/rules/new_lines.py

```diff
@@ -30,24 +33,27 @@
      type: unix
 """
 
+from os import linesep
 
 from yamllint.linter import LintProblem
 
 
 ID = 'new-lines'
 TYPE = 'line'
-CONF = {'type': ('unix', 'dos')}
+CONF = {'type': ('unix', 'dos', 'platform')}
 DEFAULT = {'type': 'unix'}
 
 
 def check(conf, line):
+    if conf['type'] == 'unix':
+        newline_char = '\n'
+    elif conf['type'] == 'platform':
+        newline_char = linesep
+    elif conf['type'] == 'dos':
+        newline_char = '\r\n'
+
     if line.start == 0 and len(line.buffer) > line.end:
-        if conf['type'] == 'dos':
-            if (line.end + 2 > len(line.buffer) or
-                    line.buffer[line.end:line.end + 2] != '\r\n'):
-                yield LintProblem(1, line.end - line.start + 1,
-                                  'wrong new line character: expected \\r\\n')
-        else:
-            if line.buffer[line.end] == '\r':
-                yield LintProblem(1, line.end - line.start + 1,
-                                  'wrong new line character: expected \\n')
+        if line.buffer[line.end:line.end + len(newline_char)] != newline_char:
+            yield LintProblem(1, line.end - line.start + 1,
+                                'wrong new line character: expected {}'
+                                .format(repr(newline_char).strip('\'')))
```

Comment:
The `.strip()` removes the apostrophes `'` from the beginning and end of the `newline_char`. This is consistent with the old messages.

We could lose the `.strip()` (making the code a little less complex) and I personally prefer the message with the apostrophe. This breaks `tests/test_cli.py, line 678` and would require the apostrophes to be added there to, so it's probably not worth it.

Reply:
I agree with your point of view. If the rule was to be created now, maybe I would use the apostrophes too. But let's avoid breaking changes, in case an automated tool relies on an exact string to be output by yamllint.